### ISSUE TITLE
Tasks created or updated outside of TB are not refreshed

### ISSUE THREAD

**Issue by alexivkin** [kewisch/gdata-provider#72]
### Describe the problem and steps to reproduce it:
Creating or completing a task using google.com does not add or change it in the Tasks tab. Refreshing the tasks "calendar" in TB or waiting the default 60 minutes does not help.

Attempting to create the same task in TB creates a duplicate on google.com. 
Attempting to complete the already completed task in TB shows a dialog "the item has recently been changed on the server..." allowing to discard and reload changes  or submitting the changes anyway.

This only happens when the tasks are completed outside of Thuderbird. For tasks completed in the TB the updates show up on google.com correctly.

Disabling and reenabling the "tasks" calendar does not help. 
Completely removing it and recreating does help refresh data in TB, but the problem is still there for the next item created or changed outside of TB.

### What did you expect to happen?
Thunderbird should receive updates for when a task is completed or created outside of it.

### Anything else we should know?

The logs show sync happening as it should
```
Request GET https://www.googleapis.com/tasks/v1/lists/[UIDHERE]/tasks?maxResults=1000 responded with HTTP 200
console.log: Lightning: [calGoogleCalendar] Parsing 9 received tasks
console.log: Lightning: CalDAV: recv:
<D:multistatus>
 <D:sync-token>/caldav/v2/[EMAILHERE]/events/sync/[TOKENHERE]</D:sync-token>
</D:multistatus>

console.log: Lightning: CalDAV: New webdav-sync Token: /caldav/v2/[EMAILHERE]/events/sync/[TOKENHERE]
console.log: Lightning: aChangeLogListener=[xpconnect wrapped calIGenericOperationListener]
calendarURI=https://apidata.googleusercontent.com/caldav/v2/[EMAILHERE]/events/
iscached=true
this.mQueuedQueries.length=0
console.log: Lightning: [calCachedCalendar] replayChangesOn finished.
console.log: Lightning: [calCachedCalendar] sync queue empty.
console.log: Lightning: [calGoogleCalendar] Request GET https://www.googleapis.com/tasks/v1/lists/[UIDHERE]/tasks?maxResults=1000 responded with HTTP 200
console.log: Lightning: [calGoogleCalendar] Parsing 8 received tasks
console.log: Lightning: [calGoogleCalendar] Last sync date for My Tasks(tasks) is now: .... UTC isDate=0 nativeTime=....
console.log: Lightning: [calCachedCalendar] replayChangesOn finished.
```
TB Addon **v68.2.1**

**Comment by Sebastian1969** [comment]
I experience the same issue with a Google account and tasks.

Generating a new task in Thunderbird: OK, synchronized and visible in Google Android App and Online
Deleting an existing task in Thunderbird: OK, synchronized and visible in Google Android App and Online

Deleting an existing task in Google Android App: **NOT** synchronized, item still visible in Thunderbird.

**Comment by mat4444** [comment]
I have the same issue as well and I would add this too:

Mark a task as completed in Thunderbird: OK,  synchronized in Google Task and Google web Agenda
Mark a task as completed in Google Task app or Google Agenda: **NOT** synchronized with Thunderbird/Lightning

I have these version of Thunderbird: 68.10.0 (64 bits) and Lightning: 68.10.0

**Comment by rebel2k** [comment]
Same as @mat4444, when I mark a task as completed in Google Tasks, the status is never updated in TB.

**Comment by rebel2k** [comment]
Anything I can provide to help with looking at that issue?

**Comment by Sebastian1969** [comment]
Philipp, I know there are many things to do. But this is an issue that really calls for a solution as tasks synchronization is not possible via multiple devices because of this behavior. Could you please have a look for confirmation at least? I think here are people willing to help but we need some interaction with you.

**Comment by kewisch** [comment]
Hi Sebastian, apologies I haven't been able to track this down. Does this also happen with Thunderbird 78 and the latest version of the provider? Most reports I've seen in the issue here are around Thunderbird 68.

**Comment by Sebastian1969** [comment]
Hi Philipp,
many thanks for your reply and no worries. Yes indeed, I am running Thunderbird 78.3.2 (64-Bit) and Provider for Google Calendar 78.0.1 - **and the issue is the same, unfortunately.**

One comment: In my description I just referred to "generate" and "delete" tasks.  mat4444 even added "mark as completed". [https://github.com/kewisch/gdata-provider/issues/72#issuecomment-663964362](url). I guess all these use cases need to be considered.

Would be really great to get that fixed :-) (and I would assume it is more a Google API issue than a programming issue on your side).

Best regards Sebastian

**Comment by mat4444** [comment]
Hello,
I have the same version as Sebastian, and I confirm the issue still exists with TB 78.

**Comment by Sebastian1969** [comment]
I am sorry but I moved to Evolution for the time being. Evolution works exactly as expected in terms of the tasks management in sync with Google Tasks (but there are other drawbacks). However, still happy to help.

**Comment by TobiasLaa** [comment]
Anything new on this issue? 
I'm having the same problem with TB 78.4.2 and Provider for Google calendar 78.0.1.
I use the tasks mainly on my mobile, but completed or moved tasks are still with old values in TB.

**Comment by rebel2k** [comment]
Yep, would like to have some news too... 👍

**Comment by Sebastian1969** [comment]
And I would move back from Evolution to Thunderbird immediately.

**Comment by TobiasLaa** [comment]
I've added an other extension to open native google calendar, but this should be only a workaroud

**Comment by Sebastian1969** [comment]
@TobiasLaa, I agree. Well, I know this extension "Open Google Calendar". But I don't like to use it because I prefer to have an email client (+ extensions) that works as expected and doesn't provide "browser integration". If I am forced to use the "Open Google Calendar" extension, then gdata-provider is completely useless and IMHO does not need to be developed further. Sorry for the clear words but an extension is useless if not working properly.

**Comment by TobiasLaa** [comment]
@Sebastian1969 , as I wrote.. Theo "Open Google Calendar" extension only a workaround. I also would have TB with a working gdata-provider....

**Comment by Sebastian1969** [comment]
@TobiasLaa, yes I know - we have exactly the same understanding. Sorry if my words are misleading.

**Comment by mac-mac** [comment]
Hello,
I'm having same issue.
When I delete a task from within Thunderbird (with Google Calendar Provider) I can see right away on Google Calendar (calendar.google.com) that it disappears. However, when I delete a task directly from Google Calendar, the provider seems to have problem syncing and a task in Thunderbird just doesn't get deleted.

Thunderbird 78.6.1 (64 bit)
Google Calendar Provider 78.0.1

**Comment by kewisch** [comment]
Fixed on legacy branch in dad61c9b8ba813318d3bcae749cda8b2d71c9fc2
